### DISCUSSION TITLE
issue: RHF watch method triggers if field of another object field is changed

### DISCUSSION THREAD

**Discussion by fligge37** [react-hook-form/react-hook-form#7271] [Q&A]
### Version Number

7.17.5

### Codesandbox/Expo snack

https://codesandbox.io/s/young-platform-8thsm?file=/src/App.tsx

### Steps to reproduce

1. Go to Sanbox
2. Type any value in Person1 and Person 2 input
3. Watch the terminal output
(4. Try the same without Line 24 person2FirstName)

### Expected behaviour

I have a use case where i have two interfaces for my form type. I want to watch for any change of the first object as a whole, and watch a specific value from the other. If I only watch for any changes of the first object, the useEffect in my example work as expected bc. it will only trigger on person1 change. If I also watch for some value of the second object (CodeSandBox: Line 24 person2FirstName), the useEffect will trigger on any change of person2 aswell.

I would expect that this watch method doesnt trigger if any other value of my form changes.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
> watch result is optimised for render phase instead of useEffect's deps, to detect value update you may want to use an external custom hook for value comparison.

https://react-hook-form.com/api/useform/watch

**Reply by fligge37** [reply]
Thank you for your insanely fast reply!
I now use the example provided in the watch docu that you mentioned and check the name for the correct type to detect the change on person1.

Like so: 

```tsx
React.useEffect(() => {
    const subscription = watch((value, { name }) => {
      if (name?.includes("person1")) {
        console.log(name);
      }
    });
    return () => subscription.unsubscribe();
  }, [watch]);
  ```
